plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java

```diff
@@ -482,4 +591,18 @@ private static void validateColumnNameForInsert(String columnName)
             throw new IllegalArgumentException("Column name must not contain '$' or '.' for INSERT: " + columnName);
         }
     }
+
+    private static MongoColumnHandle getPageSinkIdColumn(List<String> otherColumnNames)
```

Comment:
I would rename the method to `buildPageSinkIdColumn` and change the argument to `Set<String>`.